test/functional/foreman_discovery/concerns/hosts_controller_extensions_test.rb

```diff
@@ -12,11 +12,11 @@ class HostsControllerTest < ActionController::TestCase
             'discovered_host' => { 'architecture_id' => architecture.id }
           }
 
-          @request.stubs(:path).
-            returns(architecture_selected_discovered_hosts_path)
-
+          @request.stubs(:path).returns(architecture_selected_discovered_hosts_path)
           post :architecture_selected, discovered_host_params, set_session_user
-          assert assigns('architecture'), architecture
+
+          # normally we would test response.body but in this case it contains OS entries
+          assert_equal architecture, @controller.instance_variable_get("@architecture")
```

Comment:
in Rails 5 they made functional tests equal to integration tests, we should perhaps test that response contains architecture.name or something and what was the response code

Reply:
Agree but see the comment - the response actually does not contain architecture but operating systems associated with this architecture. It's confusing. I could however do some factorygirl magic to create some associations and test that if you insist.

Reply:
Ok changed it to integration test.